### PR TITLE
docs: migrate docs to tfplugindocs

### LINKED ISSUE #38: ci: generate docs automatically with tfplugindocs tool
https://github.com/hashicorp/terraform-plugin-docs

### PR DESCRIPTION
**Pull request by nint8835**
# Changes
Resolves #38 

This PR migrates documentation from being manually written to auto-generating it via the `tfplugindocs` tool. As part of doing so, it adds descriptions to all schema fields & objects, where relevant.

I've also done a review of all existing docs and done some general cleanup to make them more consistent & readable.

# Work remaining

- [x] Confirm there's no way to omit fields from the docs (to omit `type` on channels and the individual permission bits on the permissions data source) (https://github.com/hashicorp/terraform-plugin-docs/issues/274)
- [x] Confirm there's no way to set the provider description on the terraform provider SDK, to avoid needing to create a custom index.md template to populate the provider description
- [x] Document `discord_system_channel` resource
- [x] Document message embed fields
- [x] Give a once-over of all existing docs, revising where necessary

### CODE REVIEW

**Comment by Lucky3028** [comment]
omg
how wonderful pr this is!!!
thank you!!!

**Comment by Lucky3028** [comment]
note: need to install tfplugindocs on CI

**Comment by nint8835** [comment]
The markdown linter is failing as well - can I remove it, now that the markdown is auto-generated?

**Comment by Lucky3028** [comment]
@nint8835 Sure! You can do that.

File: discord/resource_discord_channel.go [hunk 1]
```diff
@@ -15,20 +15,24 @@ import (
...
+			Description: "The ID of the channel.",
 		},
-		"category": {
-			Type:     schema.TypeString,
-			Optional: true,
+		"channel_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "The ID of the channel.",
 		},
```
Comment (Lucky3028) [inline discord/resource_discord_channel.go:31]: Why did you remove `category`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,3 +24,11 @@ jobs:
...
       - name: Test
         run: go test -v ./...
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v3
+      - name: Generate docs
+        run: go generate ./...
+      - name: Check for documentation changes
+        run: |
+          git diff --compact-summary --exit-code || \
+            (echo; echo "Unexpected difference in directories after code generation. Run 'go generate ./...' command and commit."; exit 1)
```
Comment (Lucky3028) [inline .github/workflows/ci.yml:34]:
```suggestion
        run: |
          result=0
          git diff --compact-summary --exit-code || result=$?
          if [ ! "$result" -eq "0" ]; then
            echo
            echo "Unexpected difference in directories after code generation. Run 'go generate ./...' command and commit."
            exit 1
          fi
```

**Review by Lucky3028 — CHANGES_REQUESTED** [review]

**Comment by Lucky3028** [comment]
Otherwise LGTM.

File: discord/resource_discord_channel.go (hunk 1 above)
Comment (nint8835) [reply discord/resource_discord_channel.go:31]: A few lines below, `category` is added if the channel type is not category, so this particular category definition is only used in category channels https://github.com/Lucky3028/terraform-provider-discord/blob/10f2774e5cf052f7b6c390ccc5b914c66c92b61a/discord/resource_discord_channel.go#L69-L74

In hindsight, this is technically a breaking change from a schema perspective and so I'm fine with reverting this if desired, but there is a check that would cause it to error if someone had provided this option on a category channel before https://github.com/Lucky3028/terraform-provider-discord/blob/10f2774e5cf052f7b6c390ccc5b914c66c92b61a/discord/resource_discord_channel.go#L94-L98

Comment (Lucky3028) [reply discord/resource_discord_channel.go:31]: Alright, we are on the same page. You don’t have to revert.
I couldn't find any changes similar to this one, are there others?

Comment (nint8835) [reply discord/resource_discord_channel.go:31]: Nope, I believe this is the only structural schema change

Comment (Lucky3028) [reply discord/resource_discord_channel.go:31]: Thanks!

**Review by Lucky3028 — APPROVED** [review]
